src/XRP/reliable-submission-xrp-client.ts

```diff
@@ -19,6 +19,140 @@ async function sleep(milliseconds: number): Promise<void> {
 export default class ReliableSubmissionXrpClient implements XrpClientDecorator {
   public constructor(
     private readonly decoratedClient: XrpClientDecorator,
+    readonly network: XrplNetwork,
+  ) {}
+
+  public async getBalance(address: string): Promise<BigInteger> {
+    return this.decoratedClient.getBalance(address)
+  }
+
+  public async getPaymentStatus(
+    transactionHash: string,
+  ): Promise<TransactionStatus> {
+    return this.decoratedClient.getPaymentStatus(transactionHash)
+  }
+
+  public async send(
+    amount: string | number | BigInteger,
+    destination: string,
+    sender: Wallet,
+  ): Promise<string> {
+    return this.sendWithDetails({
+      amount,
+      destination,
+      sender,
+    })
+  }
+
+  public async sendWithDetails(
+    sendMoneyDetails: SendXrpDetails,
+  ): Promise<string> {
+    const { sender } = sendMoneyDetails
+    const ledgerCloseTimeMs = 4 * 1000
+
+    // Submit a transaction hash and wait for a ledger to close.
+    const transactionHash = await this.decoratedClient.sendWithDetails(
+      sendMoneyDetails,
+    )
+    await sleep(ledgerCloseTimeMs)
+
+    // Get transaction status.
+    let rawTransactionStatus = await this.getRawTransactionStatus(
+      transactionHash,
+    )
+    const { lastLedgerSequence } = rawTransactionStatus
+    if (!lastLedgerSequence) {
+      return Promise.reject(
+        new Error(
+          'The transaction did not have a lastLedgerSequence field so transaction status cannot be reliably determined.',
+        ),
+      )
+    }
+
+    // Decode the sending address to a classic address for use in determining the last ledger sequence.
```

Comment:
nit: but should we shift these to `/* */` style comments as opposed to a new `//` for each line?

Reply:
I tend to prefer `//` inside of functions.